lib/verify.js

```diff
@@ -25,10 +25,14 @@ function verifySignature(req, opts, cb) {
 	try {
 		// parse the request
 		var parsed = httpSignature.parseRequest(req, opts);
+
+		//Use different verification methods depending if the algorithm is HMAC or not
+		var is_hmac = parsed.algorithm.toLowerCase().indexOf('hmac') === 0;
+		verify_func = (is_hmac) ?  httpSignature.verifyHMAC : httpSignature.verifySignature;
```

Comment:
@aisipos This needs to be `var verifyFunc`. Again, nit-picking on the casing, but really need to avoid leaking a global variable.